raiden/network/rpc/client.py

```diff
@@ -460,7 +467,7 @@ def __init__(
         self.address = discovery_address
         self.proxy = proxy
         self.client = jsonrpc_client
-        self.startgas = startgas
+        self.startgas = get_block_gaslimit(self.client)
```

Comment:
This way you instantiate `startgas` only once. That is during the construction of the object. This won't work since we want to have an adjustable gas limit as time progresses and new blocks are mined.

Same comments goes for all other places where you do this.

Also perhaps if we are to use the blockgas limit it's proper to rename this variable from `startgas` to `blockgaslimit`.